python/tvm/relay/frontend/tensorflow.py

```diff
@@ -606,6 +606,11 @@ def _impl(inputs, attr, params):
 def _reshape():
     def _impl(inputs, attr, params):
         pop_node = inputs.pop(1)
+
+        # We use reshape_like directly to deal with dynamic shape.
+        if isinstance(pop_node, tvm.relay.expr.Call):
+            return _op.reshape_like(inputs[0], pop_node.args[0])
```

Comment:
Doesnt this only work when reshape is directly preceded by a shape operation? Is that something that we can assume will always be true in TF dynamic reshape calls?

Reply:
We only need to special handle reshape like situation. For normal reshape, we can still use the existing logic.

Reply:
I might need to consider a bit more about other more complicated cases.

Reply:
@jwfromm I added a check to only allow reshape_like to directly follow shape operator for symbolic input shape. This should cover most common cases for reshape like. Covering more general cases can require significant change for frontend infer_type system, and we can improve this gradually.